### PR TITLE
e2e: Use `unlockWallet` helper consistently

### LINKED ISSUE #21914: Refactor e2e tests to use `unlockWallet` consistently
### What is this about?

Our e2e tests always repeat the implementation for unlocking the wallet, even though there is an helper function to do that.

Moreover, the `unlockWallet` helper function should also wait for the login to actually end, so that test cases don't have to manually delay the execution for an arbitrary number of seconds.

### Scenario

_No response_

### Design

_No response_

### Technical Details

Some of our test cases open external pages in the browser to test dapp features, and these tests take action on these pages right after the login. Some of them explicitly wait some time (with driver.delay(1000);), some others don't wait at all: this was ok-ish when the key derivation and vault decryption was fast, because the login operation used to end before the external pages were loaded, but now that the decryption is much slower this assumption is wrong because this happens:

1. The test case inputs the password, then clicks on login
2. The test case immediately opens an external page
3. The test case immediately executes an action on the test dapp
4. The pop-up (any pop-up screen, like the one to sign for example) will open, but instead of showing the expected screen, it will show the login screen to input the password, as the decryption is still happening on the main initial view.
5. The test fails with a timeout, as the screen expected by the test will never show up.

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by mikesposito**
## **Description**

This PR refactors e2e tests to use a single shared helper function to unlock the wallet, instead of implementing it in each of them. 

This function also waits for the login to complete, before giving the control flow back to the test.

## **Related issues**

Fixes: #21914 

## **Manual testing steps**

n/a - e2e tests should work the same

## **Screenshots/Recordings**

n/a
### **Before**

n/a
### **After**

n/a
## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/helpers.js [hunk 1]
```diff
@@ -710,9 +709,15 @@ async function waitForAccountRendered(driver) {
   );
 }
 
-async function unlockWallet(driver) {
+async function unlockWallet(driver, waitLoginSuccess = true, navigate = true) {
```
Comment (mcmire) [inline test/e2e/helpers.js:712]: Nit: Perhaps we want to use an options argument for `waitLoginSuccess` and `navigate` so if we see `unlockWallet(driver, false)`, we know what the `false` is for (and also allows `navigate` to be specified, say, without having to specify `waitLoginSuccess`).

**Review by mcmire — DISMISSED** [review]
One nit, but this looks good regardless.

File: test/e2e/helpers.js (hunk 1 above)
Comment (mikesposito) [reply test/e2e/helpers.js]: Good idea! This also makes easier to extend options in the future, refactored in https://github.com/MetaMask/metamask-extension/pull/21915/commits/aad0de0fe8e81429f03c4a415246487f4c2fcde6

**Review by mcmire — APPROVED** [review]
Looks good!

**Review by HowardBraham — APPROVED** [review]
Awesome, I had been putting in `unlockWallet` a few tests at a time, but you took it to the next level!

**Review by bowensanders — APPROVED** [review]
LGTM 👍

**Pull request merged**